### PR TITLE
Fix: cygwin compilation of BMDA

### PR DESCRIPTION
**Pull request by hydra**
Prior to this the following command gave many errors.
```
make PROBE_HOST=hosted HOSTED_BMP_ONLY=0
```

This fixes are by 'dragonmux' on Discord.  tested here on Windows10/Cygwin64.

### CODE REVIEW

**Review by dragonmux — CHANGES_REQUESTED** [review]
There are a couple of code changes that we'd like to see made before we can merge this. The patches to fix cygwin compilation are greatly appreciated though!

We're aware the commits were rather hastily thrown together, so when you get a moment please review our [contribution guidelines for commit messages](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md#commit-message-guidelines) and adjust the commit messages.

**Review by dragonmux — APPROVED** [review]
LGTM, we'll merge this when the builds complete.

File: src/platforms/hosted/gdb_if.c
Comment (dragonmux) [inline src/platforms/hosted/gdb_if.c:26]: It seems this re-ordering blew the build for Linux and may have subtly done so for windows not-cygwin, unsure on fruitos.

Given this, we suggest `#ifndef __CYGWIN__` guarding this version of the include statement, and then `#ifdef __CYGWIN__` guarding the one added below

**Review by dragonmux — CHANGES_REQUESTED** [review]

File: src/platforms/hosted/gdb_if.c
Comment (hydra) [reply src/platforms/hosted/gdb_if.c:26]: Done, let's see if the build passes now. 😄

**Comment by hydra** [comment]
Rebased on `main` too.

**Review by dragonmux — APPROVED** [review]

**Comment by dragonmux** [comment]
Looks good, merging. Thanks for the contribution!

**Comment by hydra** [comment]
> Looks good, merging. Thanks for the contribution!

You're welcome, though I feel I didn't do much other than try your suggested fixes 😄 😄 😄   You clearly know the code-base well.  Thanks again for the assistance in getting it up and running here, much appreciated.

**Pull request merged**
